core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.ts

```diff
@@ -0,0 +1,394 @@
+  acceptedQuestions: number;
+  acceptedQuestionsWithoutEdits: number;
+}
+
+interface QuestionReviewStats extends Stat {
+  reviewedQuestions: number;
+  acceptedQuestions: number;
+}
+
+@Component({
+  selector: 'contributor-stats',
+  templateUrl: './contributor-stats.component.html',
+  styleUrls: []
+})
+export class ContributorStatsComponent {
+  @Input() type!: string;
+  @ViewChild('dropdown', {'static': false}) dropdownRef!: ElementRef;
+  @ViewChild('mobileDropdown', {'static': false}) mobileDropdownRef!:
+    ElementRef;
+
+  dropdownShown: boolean = false;
+  mobileDropdownShown: boolean = false;
+  selectedContributionType: string | undefined = '';
+  username: string = '';
+  ITEMS_PER_PAGE: number = 5;
+
+  userCanReviewTranslationSuggestions: boolean = false;
+  userCanReviewQuestionSuggestions: boolean = false;
+  userCanSuggestQuestions: boolean = false;
+
+  COLUMNS = {
+    translationContribution: {
+      months: 'Months',
+      topicNames: 'Topic Names',
+      acceptedCards: 'Accepted Cards',
+      acceptedWordCount: 'Accepted Word Count'
+    },
+    translationReview: {
+      months: 'Months',
+      topicNames: 'Topic Names',
+      reviewedCards: 'Reviewed Cards',
+      reviewedWordCount: 'Reviewed Word Count',
+      acceptedCards: 'Accepted Cards',
+      acceptedWordCount: 'Accepted Word Count'
+    },
+    questionContribution: {
+      months: 'Months',
+      topicNames: 'Topic Names',
+      acceptedQuestions: 'Accepted Questions',
+      acceptedQuestionsWithoutEdits: 'Accepted Questions Without Edits'
+    },
+    questionReview: {
+      months: 'Months',
+      topicNames: 'Topic Names',
+      reviewedQuestions: 'Reviewed Questions',
+      acceptedQuestions: 'Accepted Questions'
+    }
+  };
+
+  translationContributionOption: Option = {
+    displayName:
+      AppConstants.CONTRIBUTION_STATS_TYPES
+        .TRANSLATION_CONTRIBUTION.DISPLAY_NAME,
+    contributionType:
+     AppConstants.CONTRIBUTION_STATS_TYPES.TRANSLATION_CONTRIBUTION.NAME
+  };
+
+  translationReviewOption: Option = {
+    displayName:
+      AppConstants.CONTRIBUTION_STATS_TYPES.TRANSLATION_REVIEW.DISPLAY_NAME,
+    contributionType:
+     AppConstants.CONTRIBUTION_STATS_TYPES.TRANSLATION_REVIEW.NAME
+  };
+
+  questionContributionOption: Option = {
+    displayName:
+      AppConstants.CONTRIBUTION_STATS_TYPES.QUESTION_CONTRIBUTION.DISPLAY_NAME,
+    contributionType:
+     AppConstants.CONTRIBUTION_STATS_TYPES.QUESTION_CONTRIBUTION.NAME
+  };
+
+  questionReviewOption: Option = {
+    displayName:
+      AppConstants.CONTRIBUTION_STATS_TYPES.QUESTION_REVIEW.DISPLAY_NAME,
+    contributionType:
+     AppConstants.CONTRIBUTION_STATS_TYPES.QUESTION_REVIEW.NAME
+  };
+
+  options: Option[] = [
+    this.translationContributionOption
+  ];
+
+  statsData: {
+    translationContribution?: {[key: string]: PageableStats | undefined} |
+      undefined;
+    translationReview?: {[key: string]: PageableStats | undefined} | undefined;
```

Comment:
Let's just instantiate the map here and then we can also remove the `undefined` types.

Reply:
Done